File: FizzBuzz/fizzbuzz-maven/src/main/java/FizzBuzzDemo.java
public class FizzBuzzDemo {
    public static String of(int given) {
        if (given % 3 == 0 && given % 5 == 0) {
            return "FizzBuzz";
        }
        if (given % 3 == 0) {
            return "Fizz";
        }
        if (given % 5 == 0) {
            return "Buzz";
        }
        return String.valueOf(given);
    }
}

File: FizzBuzz/FizzBuzz2-maven/src/main/java/FizzBuzzDemo.kt
object FizzBuzzDemo {
    fun of(given: Int): String = when{
        isMultiple3(given) && isMultiple5(given) -> "FizzBuzz"
        isMultiple3(given) -> "Fizz"
        isMultiple5(given) -> "Buzz"
        else-> given.toString()
    }

    private fun isMultiple5(given: Int) = given % 5 == 0

    private fun isMultiple3(given: Int) = given % 3 == 0

}
